plugins/glustershd/rest.go

```diff
@@ -1,197 +1,95 @@
 package glustershd
 
 import (
+	"encoding/xml"
 	"net/http"
+	"path"
 
 	"github.com/gluster/glusterd2/glusterd2/gdctx"
 	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
 	"github.com/gluster/glusterd2/glusterd2/transaction"
-	"github.com/gluster/glusterd2/glusterd2/volume"
-	"github.com/gluster/glusterd2/pkg/errors"
+	glustershdapi "github.com/gluster/glusterd2/plugins/glustershd/api"
 
 	"github.com/gorilla/mux"
 	"github.com/pborman/uuid"
-	log "github.com/sirupsen/logrus"
+	config "github.com/spf13/viper"
 )
 
-func isVolReplicate(vType volume.VolType) bool {
-	if vType == volume.Replicate || vType == volume.Disperse || vType == volume.DistReplicate || vType == volume.DistDisperse {
-		return true
-	}
-
-	return false
-}
-
-func glustershEnableHandler(w http.ResponseWriter, r *http.Request) {
-	// Implement the help logic and send response back as below
+func selfhealInfoHandler(w http.ResponseWriter, r *http.Request) {
 	p := mux.Vars(r)
 	volname := p["name"]
+	option := p["opts"]
+	_ = option
 
 	ctx := r.Context()
 	logger := gdctx.GetReqLogger(ctx)
 
-	//validate volume name
-	v, err := volume.GetVolume(volname)
-	if err != nil {
-		if err == errors.ErrVolNotFound {
-			restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound)
+	lock, unlock := transaction.CreateLockFuncs(volname)
```

Comment:
If you are taking locks outside, you can do validations in the handler itself instead of step function